### PR TITLE
add list access to get_dotted_field_value

### PR DESCRIPTION
**Pull request by ekneg54**
* this adds simple list access to `get_dotted_field_value` helper function
* slicing is not supportet yet

### CODE REVIEW

**Comment by ekneg54** [comment]
closes #298

File: logprep/util/helper.py [hunk 1]
```diff
@@ -104,6 +104,13 @@ def add_field_to(event, output_field, content, extends_lists=False, overwrite_ou
     return False
 
 
+def _get_item(items, item):
+    try:
+        return dict.__getitem__(items, item)
+    except TypeError:
+        return list.__getitem__(items, int(item))
```
Comment (dtrai2) [inline logprep/util/helper.py:111]: looks good to me. I am not sure if there is a need for it, but it is fairly simple to extend this to get characters from a string with `str.__getitem__(items, int(item))`. As one single character might be uninteresting though you could also extend it with the slicing functionality. For that you "just" have to check if the `item` string has a `:` colon in it, split the string and pass it into the `slice(start, stop, step)` method.

File: logprep/util/helper.py [hunk 2]
```diff
@@ -124,9 +131,11 @@ def get_dotted_field_value(event: dict, dotted_field: str) -> Optional[Union[dic
...
     fields = [event, *dotted_field.split(".")]
     try:
-        return reduce(dict.__getitem__, fields)
+        return reduce(_get_item, fields)
     except KeyError:
         return None
+    except ValueError:
+        return None
     except TypeError:
         return None
```
Comment (dtrai2) [inline logprep/util/helper.py:140]: Here you also have to check for the `IndexError` and write in the `test_helper.py` file a corresponding test. Just for the cases where someone tries to access an element, that is outside of the boundaries of the collection.

**Review by dtrai2 — CHANGES_REQUESTED** [review]

**Review by dtrai2 — APPROVED** [review]
I just added some documentation and more test cases, as well as the case for IndexErrors. On my side it can be merged now, but please take a look at my changes first before doing so.

**Pull request merged**